### PR TITLE
TECH Update logger setup to inject configuration

### PR DESCRIPTION
**Pull request by lledey**
Update pretty much everything in the project:
* update circleci config to run the tests against multiple versions of node
* add prettier to the project
* use nyc instead of instanbul
* remove unused depdendencies
* remove useless package-lock.json since we use node 6 and npm at that time didn't have this feature
* update logger initialization to be able to inject the configuration (also update the typings accordingly)

### CODE REVIEW

File: .nycrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (vschoener) [inline .nycrc]: I looked what was this file online and I found this:

New York City Regional Center :: NYCRC 😄

File: index.js [hunk 2]
```diff
@@ -1,47 +1,73 @@
...
-  client.install();
-  config.streams.push(sentryStream(client));
+  return bunyan.createLogger(loggerConfig);
 }
 
-// Logger instance:
-const logger = bunyan.createLogger(config);
+const logger = init(defaultConfig);
 
 module.exports = logger;
```
Comment (vschoener) [inline index.js]: As it's a breaking change, maybe we should force projects to only use the init method and not initialize the logger internally?

```js
module.exports = {
  init
} 
```

File: .nvmrc [hunk 3]
```diff
@@ -1 +1 @@
-4.8.4
\ No newline at end of file
+6.0.0
```
Comment (vschoener) [inline .nvmrc]: It's not possible to use a recent version due to projects using an old one right?

Comment (ygrandmaitre) [inline .nvmrc]: I think you should use the last version of node 6 as like the one used in Circleci

**Review by ygrandmaitre — APPROVED** [review]
Thank, good work.
LGTM

File: index.js (hunk 2 above)
Comment (lledey) [reply index.js]: Let's do it progressively until we are sure we will use the sentry releases everywhere

File: .nvmrc (hunk 3 above)
Comment (lledey) [reply .nvmrc]: Yes, we should target the lowest supported version

Comment (lledey) [reply .nvmrc]: I think we should use the lowest possible instead. I updated the circleci config to use node v6.11.1 (6.0.0 is not available...)

**Pull request merged**